apps/webapp/web/static/js/docs/index.js

```diff
@@ -24,24 +25,42 @@ class Index extends React.Component {
         return response.json();
       })
       .then(list => {
-        this.setState({ docs: list });
+        this.setState({ docs: list.data });
+      });
+
+    fetch(window.location.origin + '/docs/use_case_examples.md')
+      .then(response => {
+        if (response.status >= 400) {
+          throw new Error("Bad response from server");
+        }
+        return response.json();
+      })
+      .then(doc => {
+        this.setState({ file: doc.data });
       });
   }
 
   changeDocument (e) {
     e.preventDefault();
-    this.setState({
-      path: e.target.getAttribute('data-url')
-    })
+    fetch(window.location.origin + '/docs/' + e.target.getAttribute('data-filename'))
+      .then(response => {
+        if (response.status >= 400) {
+          throw new Error("Bad response from server");
+        }
+        return response.json();
+      })
+      .then(doc => {
+        this.setState({ file: doc.data });
```

Comment:
In this case you can DRY this fetch code given that you are repeating the same operations: fetching and setting the "file" property in the state. You could have a function fetchFile with an url as param or something similar..